py/desispec/spectra.py

```diff
@@ -979,7 +1028,7 @@ def stack(speclist):
         _remove_tile_keywords(headers)
 
     sp = Spectra(bands, wave, flux, ivar,
-        mask=mask, resolution_data=rdat,
+        mask=mask, model=model, resolution_data=rdat,
```

Comment:
I think you found all the cases where spectral slicing and stacking needs to manipulate the models just like it does for flux, ivar, etc., but please add checks for that to `desispec.test.test_spectra` in the `test_slice` and `test_stack` functions.